src/ReactiveUI/ViewLocator.cs

```diff
@@ -93,31 +95,27 @@ public DefaultViewLocator(Func<string, string> viewModelToViewFunc = null)
         public IViewFor ResolveView<T>(T viewModel, string contract = null)
             where T : class
         {
-            var view = this.AttemptViewResolutionFor(viewModel.GetType(), contract);
+            var view = GetCandidateViewModelTypes(viewModel)
```

Comment:
Yeah, benchmarkdotnet could tell us precisely what's going on in terms of memory allocs. Only problem is, we don't really have the infrastructure in place to run it in the solution as it stands. You could easily add a quick check, but it might be throwaway code until we formally decide how to incorporate perf tests into the solution.